scaleway/data_source_tem_domain.go

```diff
@@ -0,0 +1,77 @@
+package scaleway
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	tem "github.com/scaleway/scaleway-sdk-go/api/tem/v1alpha1"
+	"github.com/scaleway/scaleway-sdk-go/scw"
+)
+
+func dataSourceScalewayTemDomain() *schema.Resource {
+	// Generate datasource schema from resource
+	dsSchema := datasourceSchemaFromResourceSchema(resourceScalewayTemDomain().Schema)
+
+	// Set 'Optional' schema elements
+	addOptionalFieldsToSchema(dsSchema, "name", "region")
+
+	dsSchema["name"].ConflictsWith = []string{"id"}
+	dsSchema["id"] = &schema.Schema{
```

Comment:
Use `domain_id` instead of `id`.
Having a non regional id could trigger a state change when datasource will set it with a regional id